test/syntax.jl

```diff
@@ -3187,6 +3187,22 @@ end
     end
     @test err == 5 + 6
     @test x == 1
+
+    x = 0
+    try
+    catch
+    else
+        x = 1
+    end
+    @test x == 1
```

Comment:
```suggestion
    @test x == 0
```
Isn't `x == 1` the current behavior?

Reply:
No, here this is correct because `x` belongs to the outer scope

Reply:
You are right that `x==1` is also the current behavior. It is now a [local scope](https://docs.julialang.org/en/v1/manual/variables-and-scoping/#local-scope) like `try` blocks when they had no scope like `begin` blocks.

Reply:
Right. sorry for the noise!